library/common/api/external.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <string>
+
+namespace Envoy {
+namespace Api {
+namespace External {
+
+/**
+ * Register an external runtime API for usage (most likely in extensions).
+ * NOTE: This is a proof of concept implementation and a HACK. Registration is NOT thread-safe.
```

Comment:
Do we intend to fix this before merging? Seems like both of these should be thread-safe now that we've proven it works